sklearn/ensemble/tests/test_forest.py

```diff
@@ -306,11 +343,7 @@ def test_probability(name):
     check_probability(name)
 
 
-def check_importances(name, criterion, dtype, tolerance):
-    # cast as dype
-    X = X_large.astype(dtype, copy=False)
-    y = y_large.astype(dtype, copy=False)
-
+def check_importances(name, criterion, tolerance, X, y):
```

Comment:
I'd revert this change and leave the casting of the dataset here.

Reply:
following previous comment I moved the X, y definition into check_importances and removed X, y from its signature check_importances(name, criterion, tolerance)
do you suggest to go back to previous state? check_importances(name, criterion, tolerance, X, y)